### PR TITLE
Added logging to config generator and reduced computational and NPATH complexity

### PR DESCRIPTION
**Pull request by Xylonity**
The concept of having multiple methods for each explosion is cool, but that increases duplication since they essentially contain the same logic (the two functional ones so far). I haven't modified anything of that anyways tho.

I refactored some of the code into separate methods to reduce the number of processing cycles and thus lower the NPATH complexity. I haven't modified any original comments; I only added volatile constants as arguments in some LOGGER instances to prevent errors when writing data to the dedicated log file. And I also added logs in the configuration generator because it's better to avoid throwing empty exceptions.

### CODE REVIEW

File: src/main/java/net/superkat/explosiveenhancement/ExplosiveHandler.java [hunk 1]
```diff
@@ -13,29 +13,35 @@
...
                 case WATER -> spawnUnderwaterExplosionParticles(world, x, y, z, power, didDestroyBlocks, isImportant);
                 case WIND -> spawnWindExplosionParticles(world, x, y, z, power, didDestroyBlocks, isImportant); //unused for now
             }
 
-            if(config.debugLogs) { LOGGER.info("ExplosiveHandler finished!"); }
+            if (config.debugLogs) { LOGGER.info("ExplosiveHandler finished!"); }
+        }
+    }
+
+    private static void logDebugInfo() {
```
Comment (Superkat32) [inline src/main/java/net/superkat/explosiveenhancement/ExplosiveHandler.java:32]: This method will likely be replaced with a helpful YACL tooltip in the mod's actual config screen that will be much more noticeable for users. TLDR: If the particle visibility is not set to "ALL", an in-game text above all the config options will appear mentioning what's going on(This is the most common issue I receive so giving the user a hint at least seems beneficial).

File: src/main/java/net/superkat/explosiveenhancement/ExplosiveHandler.java [hunk 2]
```diff
@@ -81,7 +87,8 @@ public static boolean particlesAreWindGust(ParticleEffect particle, ParticleEffe
     public static void spawnExplosionParticles(World world, double x, double y, double z, float power, boolean didDestroyBlocks, boolean isImportant) {
         power = config.dynamicSize ? power : 4;
         y = config.attemptBetterSmallExplosions && power == 1 ? y + config.smallExplosionYOffset : y;
-        isImportant = isImportant || config.alwaysShow;
+        isImportant |= config.alwaysShow;
```
Comment (Superkat32) [inline src/main/java/net/superkat/explosiveenhancement/ExplosiveHandler.java:90]: I didn't know you could do that!

File: src/main/java/net/superkat/explosiveenhancement/config/ExplosiveConfig.java [hunk 3]
```diff
@@ -47,20 +48,19 @@ public class ExplosiveConfig {
...
         if(file.exists()) {
-            try (BufferedReader fileReader = new BufferedReader(
-                    new InputStreamReader(new FileInputStream(file), StandardCharsets.UTF_8)
-            )) {
+            try (BufferedReader fileReader = new BufferedReader(new InputStreamReader(new FileInputStream(file), StandardCharsets.UTF_8))) {
                 config = GSON.fromJson(fileReader, ExplosiveConfig.class);
             } catch (IOException e) {
-                throw new RuntimeException("Could not load Explosive Config: ", e);
+                // Throwing raw exception types should be avoided, so a log entry is used instead
+                ExplosiveEnhancement.LOGGER.error("[Explosive Enhancement]: Could not load Explosive Config. Using default configuration.", e);
```
Comment (Superkat32) [inline src/main/java/net/superkat/explosiveenhancement/config/ExplosiveConfig.java:63]: Nice catch!

File: src/main/resources/assets/explosiveenhancement/lang/es_es.json
Comment (Superkat32) [inline src/main/resources/assets/explosiveenhancement/lang/es_es.json:1]: Thank you! :)

**Review by Superkat32 — APPROVED** [review]

**Comment by Superkat32** [comment]
Oh my goodness me! This will take me some time to get merged and actually upload the update(I'm working really hard on finishing another mod before updating the rest of my mods to 1.21, then coming back to this one), but I really appreciate it! 

I've never heard of NPATH complexity before until now, but I appreciate you telling me about it and showing me how to reduce it!

I've left some notes on things I feel are important in case you are curious(I think I accidentally made it a review, don't worry about that, that was my mistake).

**Comment by Xylonity** [comment]
If the computational complexity is too high, in large modpacks you will more often notice a delay in tick rate due to processing overload, although it really depends on what the mod implements. In your case, it might take a few extra ticks when explosions are generated if there’s an overload on the server (mc isn't exactly well optimized xd), and this is relatively disproportionate to the player's experience. The best way to reduce it is to minimize code duplication as much as possible and separate everything into individual methods, including all calculations done within them (for example, instead of calling a method in a loop, put the loop inside the method’s logic), and using newer java "functions" too

If I can be of help, count on me! :)

**Comment by Superkat32** [comment]
Hiya! Sorry for the long wait, but I really appreciate your help and your willingness to teach me new things! I'm not sure when this update will be uploaded, but I'm feeling randomly motivated to update this mod today so we'll see how far I make it.

Thanks again!

**Comment by Xylonity** [comment]
Cool! If you need something i'm willing to help :)

**Pull request merged**